### DISCUSSION TITLE
Intent-filters getting mixed up

### DISCUSSION THREAD

**Discussion by alexboulay** [ThexXTURBOXx/flutter_web_auth_2#134] [Q&A]
So my app supports Android AppLinks, so a user can open `https://example.com/some-route` and get inside the app.

Here is the `intent-filter` inside my main activity:

```
<intent-filter android:autoVerify="true">
    <action android:name="android.intent.action.VIEW" />
    <category android:name="android.intent.category.DEFAULT" />
    <category android:name="android.intent.category.BROWSABLE" />
    <data android:scheme="https" android:host="www.example.com" />
</intent-filter>
```

As you may observe, the host is specified and no path, pathPattern or pathPrefix are specific so any links can open this.

On the other hand, the `intent-filter` related to `flutter_web_auth_2` does have specificity using a pathPrefix:
```
<activity
android:name="com.linusu.flutter_web_auth_2.CallbackActivity"
android:exported="true">
    meta-data android:name="flutter_deeplinking_enabled" android:value="true" />
    <intent-filter android:label="flutter_web_auth_2" android:autoVerify="true">
        <action android:name="android.intent.action.VIEW" />
        <category android:name="android.intent.category.DEFAULT" />
        <category android:name="android.intent.category.BROWSABLE" />
        <data android:scheme="https" android:host="www.example.com" android:pathPrefix="/oauth-callback"  />
    </intent-filter>
</activity> 
```

I am wrong with this approach? It seems like even the specific url still opens my default activity and the callback activity gets ignored. The only way to get it working is removing the intent-filter inside my app or make that one specific.

Any guidance or clarification is appreciated on how I can achieve this! Thanks!

**Comment by ThexXTURBOXx** [comment]
Good question... You will probably be better off asking on StackOverflow, actually.
Intent filters with `https` as scheme have always been troublesome and many issues were opened here already about this.
All I can say is that flutter_web_auth_2 is not at fault - but still, it is not very clear on how to make them work properly.

**Comment by gdurandrexel** [comment]
I have a similar problem: from what I gathered from the code (but I'm no expert in android dev), the plugin uses an intent to open the CallbackActivity and this intent contains only a scheme (the one associated with the activity). This means we cannot use this same scheme to launch another activity, even if we specify a host in the filter, because Android will consider that both activities match the intent containing only a scheme, and it will prompt the user to choose one of the matching activities. That's what I experienced anyway.

From my understanding, for us to be able to use the same scheme for different activities, we need to also specify a host and possibly a path, etc. in the filter AND we need the plugin to open the CallbackActivity using an intent containing not only a scheme, but also a host and possibly a path, etc. Am I right? And if so, could this be implemented (i.e. have specific android parameters allowing to describe a complete fliter, with at least host, path, and maybe the other supported filter attributes)?
